gleam_stdlib/src/list.gleam

```diff
@@ -184,3 +184,59 @@ pub fn zip(l1, l2) {
     | {[x1 | rest1], [x2 | rest2] } -> [ {x1, x2} | zip(rest1, rest2) ]
   }
 }
+
+pub fn intersperse(list, elem) {
+  case list {
+    | [] -> []
+    | [x | rest] -> [x | [elem | intersperse(rest, elem)]]
+  }
+}
+
+fn do_at(list, i, count) {
+  case list {
+    | [] -> Error(NotFound)
+    | [x | rest] ->
+      case i == count {
+        | True -> Ok(x)
+        | False -> do_at(rest, i, count + 1)
+      }
+  }
+}
+
+pub fn at(list, i) {
+  do_at(list, i, 0)
```

Comment:
Rather than having a helper function we can just decrement `I` and call `at`. The function can terminate when I is less than 1

Reply:
Works for me

Reply:
Thank you